enterprise/views.py

```diff
@@ -988,19 +980,77 @@ class ProgramEnrollmentView(NonAtomicView):
         'enrolled_in_course_and_paid_text': _('enrolled'),
         'enrolled_in_course_and_unpaid_text': _('already enrolled, must pay for certificate'),
-        'expected_learning_items_header': _("What you'll learn"),
-        'view_expected_learning_items_text': _('See More'),
-        'hide_expected_learning_items_text': _('See Less'),
+        'expected_learning_items_text': _("What you'll learn"),
+        'expected_learning_items_show_count': 2,
+        'corporate_endorsements_text': _('Real Career Impact'),
+        'corporate_endorsements_show_count': 1,
+        'see_more_text': _('See More'),
+        'see_less_text': _('See Less'),
         'confirm_button_text': _('Confirm Program'),
         'summary_header': _('Program Summary'),
         'price_text': _('Price'),
         'length_text': _('Length'),
         'length_info_text': _('{}-{} weeks per course'),
         'effort_text': _('Effort'),
         'effort_info_text': _('{}-{} hours per week, per course'),
+        'level_text': _('Level'),
+        'course_full_description_text': _('About This Course'),
+        'staff_text': _('Course Staff'),
+        'close_modal_button_text': _('Close'),
         'program_not_eligible_for_one_click_purchase_text': _('Program not eligible for one-click purchase.'),
     }
 
+    @staticmethod
+    def extend_course(course):
+        """
+        Extend a course with more details needed for the program landing page.
+
+        In particular, we add the following:
+
+        * `course_image_uri`
+        * `course_title`
+        * `course_level_type`
+        * `course_short_description`
+        * `course_full_description`
+        * `course_effort`
+        * `expected_learning_items`
+        * `staff`
+        """
+        try:
+            catalog_api_client = CourseCatalogApiServiceClient()
+        except ImproperlyConfigured:
+            raise Http404
+
+        course_run_id = course['course_runs'][0]['key']
+        course_details, course_run_details = catalog_api_client.get_course_and_course_run(course_run_id)
+        if not course_details or not course_run_details:
+            raise Http404
+
+        weeks_to_complete = course_run_details['weeks_to_complete']
+        course_run_image = course_run_details['image'] or {}
+        course.update({
+            'course_image_uri': course_run_image.get('src', ''),
+            'course_title': course_run_details['title'],
+            'course_level_type': course_run_details.get('level_type', ''),
+            'course_short_description': course_run_details['short_description'] or '',
+            'course_full_description': clean_html_for_template_rendering(course_run_details['full_description'] or ''),
+            'expected_learning_items': course_details.get('expected_learning_items', []),
+            'staff': course_run_details.get('staff', []),
+            'course_effort': ungettext_min_max(
```

Comment:
should we put this effort calculation in its own function for readability? i see we're doing the same thing above.

Reply:
There's already the `ungettext_min_max` abstraction I put on top of it. We could do another but any ideas where it should live? Maybe a new Mixin class for these views?